src/components/Table/GridTable.tsx

```diff
@@ -647,11 +673,11 @@ type DiscriminateUnion<T, K extends keyof T, V extends T[K]> = T extends Record<
  * column being sorted, in which case we use the GridCellContent.value.
  */
 export type GridColumn<R extends Kinded, S = {}> = {
-  [P in R["kind"]]:
+  [K in R["kind"]]:
     | string
-    | (DiscriminateUnion<R, "kind", P> extends { data: infer D }
-        ? (data: D, id: string) => ReactNode | GridCellContent
-        : (row: DiscriminateUnion<R, "kind", P>) => ReactNode | GridCellContent);
+    | (DiscriminateUnion<R, "kind", K> extends { data: infer D }
+        ? (data: D, row: GridRowKind<R, K>) => ReactNode | GridCellContent
+        : (row: GridRowKind<R, K>) => ReactNode | GridCellContent);
```

Comment:
Instead of just passing the `DiscriminateUnion` (i.e. the specific kind like `{ kind: parent, data: ParentData }` we are also really passing the `id` + `children` keys that are part of `GridDataRow`, so this is just telling the type system that, i.e. so column renderers could access `children` if they wanted.